File: crc/crc.go
package crc

const CRC_POLY_93 byte = 0x93

func CRC8_p93(crc, data byte) byte {
	crc ^= data
	var i byte = 0
	for ; i < 8; i++ {
		if (crc & 0x80) != 0 {
			crc <<= 1
			crc ^= CRC_POLY_93
		} else {
			crc <<= 1
		}
	}
	return crc
}

func CRC8_p93_2b(b1, b2 byte) byte {
	out := CRC8_p93(0, b1)
	out = CRC8_p93(out, b2)
	return out
}

File: avr-spi-uart/crc.c
#include <inttypes.h>
#include <stdbool.h>
#include "crc.h"

uint8_t const CRC_POLY_93 = 0x93;

inline uint8_t crc8_p93_next(uint8_t crc, uint8_t data) {
  crc ^= data;
  for (int8_t i = 0; i < 8; i++) {
    if ((crc & 0x80) != 0) {
      crc <<= 1;
      crc ^= CRC_POLY_93;
    } else {
      crc <<= 1;
    }
  }
  return crc;
}

inline uint8_t crc8_p93_2b(uint8_t const data1, uint8_t const data2) {
  uint8_t crc = 0;
  crc = crc8_p93_next(crc, data1);
  crc = crc8_p93_next(crc, data2);
  return crc;
}
